src/jit/rationalize.cpp

```diff
@@ -108,18 +97,17 @@ void Rationalizer::RewriteSIMDOperand(LIR::Use& use, bool keepBlk)
         addr->gtType = simdType;
         use.ReplaceWith(comp, addr);
     }
-    else if ((addr->OperGet() == GT_ADDR) && (addr->gtGetOp1()->OperIsSimdOrHWintrinsic()))
+    else if (addr->OperIs(GT_ADDR) && addr->AsUnOp()->gtGetOp1()->OperIsSimdOrHWintrinsic())
     {
-        // if we have GT_IND(GT_ADDR(GT_SIMD)), remove the GT_IND(GT_ADDR()), leaving just the GT_SIMD.
-        BlockRange().Remove(tree);
+        // If we have IND(ADDR(SIMD)) then we can keep only the SIMD node.
+        // This is a special tree created by impNormStructVal to preserve the class layout
```

Comment:
Taking the address of something that isn't addressable has to be one of the most peculiar things in the JIT IR. It will be fun to figure out to get rid of `GT_ADDR` in this case, probably there aren't many options:
1. Attach class layout to SIMD/HWINTRINSIC nodes. Probably overkill considering the rather limited need for this, call args. Though it can be regarded as a "base type" replacement which isn't unreasonable.
2. Add some kind of node that has the same role as `OBJ(ADDR(x))` without requiring addressing. Might work but I'm rather cautious about such "decorative" nodes.
3. Add class layout info to `GenTreeCall::Use`. IMO the best option.

Reply:
I agree that 3 is probably the best option.